### DISCUSSION TITLE
iOS - Rotation on different axis

### DISCUSSION THREAD

**Discussion by ipodishima** [airbnb/lottie-ios#1927] [Q&A] [answered]
Hey there,

So I've been playing lately with Lottie on iOS using Text Layers. 
I wanted to apply a rotation on a different axis than z, like y, and quickly saw it was not supported.

I managed to update the library to make it work, but I was wondering why this is not supported.
Overall, the idea would be that if there is a known limitation, I will keep the change as is for my project. Otherwise, I'd work to make it ready for a PR so that the community could enjoy it.

Thanks!

PS: this is what I wanted to achieve with the animation using a text layer
https://user-images.githubusercontent.com/968830/213674005-3f4edfd2-52f5-46c2-96c6-818c5b068aba.mp4

Here is the json file
[3d_text_test.zip](https://github.com/airbnb/lottie-ios/files/10465441/3d_text_test.zip)

**Comment by calda** [comment, marked as answer]
There's been a feature request for this for quite some time: https://github.com/airbnb/lottie-ios/issues/883

I'm not aware of any reason why this isn't supported other than the work to implement it not being prioritized. If you have an implementation that supports this, PRs are always welcome!

**Reply by ipodishima** [reply]
Yes saw that they were once even a PR but never got merged because of too many issues

I'll see what I can do then. Thanks for the prompt answer @calda
